registry/storage/filewriter.go

```diff
@@ -22,6 +23,11 @@ type fileWriter struct {
 	err    error // terminal error, if set, reader is closed
 }
 
+type bufferedFileWriter struct {
+	*fileWriter
```

Comment:
As long as the method receiver is a pointer, this should work. It will receive the address of the embedded field in the method call. Don't change `func (fw *fileWriter) readFromAt(r io.Reader, offset int64) (n int64, err error)` to `func (fw fileWriter) readFromAt(r io.Reader, offset int64) (n int64, err error)`. Only change the embed declaration here.